cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -325,6 +329,9 @@ public synchronized ProgramController start(ProgramId programId, Map<String, Str
     final ProgramController controller = runtimeInfo.getController();
     final String runId = controller.getRunId().getId();
     final String twillRunId = runtimeInfo.getTwillRunId() == null ? null : runtimeInfo.getTwillRunId().getId();
+    final RetryStrategy retryStrategy = SystemArguments.getRetryStrategy(userArguments.asMap(), programId.getType(),
```

Comment:
similar question, should we just retry forever here? This is running in the cdap system service, so it may not make sense to ever give up here.